Tests/FluentAssertions.Specs/Equivalency/DataEquivalencySpecs.cs

```diff
@@ -0,0 +1,4632 @@
+
+                var typedDataSet2 = new TypedDataSetSubclass(typedDataSet1);
+
+                typedDataSet2.CaseSensitive = !typedDataSet2.CaseSensitive;
+
+                var dataSet1 = typedDataSet1.ToUntypedDataSet();
+                var dataSet2 = typedDataSet2.ToUntypedDataSet();
+
+                // Act & Assert
+                dataSet1.Should().BeEquivalentTo(dataSet2, options => options.Excluding(dataSet => dataSet.CaseSensitive));
+            }
+
+            [Fact]
+            public void When_EnforceConstraints_does_not_match_and_property_is_not_excluded_it_should_fail()
+            {
+                // Arrange
+                var typedDataSet1 = CreateDummyDataSet<TypedDataSetSubclass>();
+
+                var typedDataSet2 = new TypedDataSetSubclass(typedDataSet1);
+
+                typedDataSet2.EnforceConstraints = !typedDataSet2.EnforceConstraints;
+
+                var dataSet1 = typedDataSet1.ToUntypedDataSet();
+                var dataSet2 = typedDataSet2.ToUntypedDataSet();
+
+                // Act & Assert
+                Action action =
+                    () =>
+                    {
+                        dataSet1.Should().BeEquivalentTo(dataSet2);
+                    };
+
+                action.Should().Throw<XunitException>();
+            }
+
+            [Fact]
+            public void When_EnforceConstraints_does_not_match_and_property_is_excluded_it_should_succeed()
+            {
+                // Arrange
+                var typedDataSet1 = CreateDummyDataSet<TypedDataSetSubclass>();
+
+                var typedDataSet2 = new TypedDataSetSubclass(typedDataSet1);
+
+                typedDataSet2.EnforceConstraints = !typedDataSet2.EnforceConstraints;
+
+                var dataSet1 = typedDataSet1.ToUntypedDataSet();
+                var dataSet2 = typedDataSet2.ToUntypedDataSet();
+
+                // Act & Assert
+                dataSet1.Should().BeEquivalentTo(dataSet2, options => options.Excluding(dataSet => dataSet.EnforceConstraints));
+            }
+
+            [Fact]
+            public void When_HasErrors_does_not_match_and_property_is_not_excluded_it_should_fail()
+            {
+                // Arrange
+                var typedDataSet1 = CreateDummyDataSet<TypedDataSetSubclass>();
+
+                var typedDataSet2 = new TypedDataSetSubclass(typedDataSet1);
+
+                typedDataSet2.TypedDataTable1.Rows[0].RowError = "Manually added error";
+
+                var dataSet1 = typedDataSet1.ToUntypedDataSet();
+                var dataSet2 = typedDataSet2.ToUntypedDataSet();
+
+                // Act & Assert
+                Action action =
+                    () =>
+                    {
+                        dataSet1.Should().BeEquivalentTo(dataSet2, config => config.ExcludingTables("TypedDataTable1"));
+                    };
+
+                action.Should().Throw<XunitException>().Which.Message.Should().Contain("HasErrors");
```

Comment:
Removing the extra block scope and splitting `Act` and `Assert` sections.

```suggestion
                // Act
                Action action = () => 
                    dataSet1.Should().BeEquivalentTo(dataSet2, config => config.ExcludingTables("TypedDataTable1"));

                // Assert
                action.Should().Throw<XunitException>().Which.Message.Should().Contain("HasErrors");
```

Reply:
I guess when I was writing it, the thought I had was that the line that assigns to `action` _isn't actually acting_, so it's disingenuous to say that it is. But, conversely, even if the initial pass through that code isn't acting, when it runs the action it is in fact executing code from that section, and in any case, what you describe is the existing convention in the codebase. I have changed it :-)